### DISCUSSION TITLE
Port to a baremetal platform

### DISCUSSION THREAD

**Discussion by ReimuNotMoe** [SquirrelJME/SquirrelJME#135] [General Questions] [answered]
Hello.

I would like to port SquirrelJME to a baremetal platform, which is a high-end MCU, with 32MB RAM and a lot of storage space. Since there's no way to make SquirrelJME work like a process on modern operating systems (and maybe only a RTOS task), its code has to be integrated into a MCU project, or has to be built as a static library (.a). Do you know how to achieve this?

It seems that I only need to integrate the `ratufacoat` part, right?

Thanks.

**Comment by XerTheSquirrel** [comment, marked as answer]
Hi!

The C port RatufaCoat can be used yes provided there is a front facing API layer for the various functions. Currently I am working on a big refactor of it, which is incomplete but is progressing fine when I am available on the weekends. This refactored version does not currently run since it is a bit in pieces, however it supports multiple different targets, it can be built into a static library, and is much more organized. For the new version I am working on you would just have to implement a front-end for any parts of it that need an implementation if you want it functional, otherwise that functionality would not be available.

As for task management, I will be supporting two threading models: cooperative and simultaneous. Cooperative being SquirrelJME managing its own threads and running consecutively, and if all cooperative threads are waiting or sleeping it could delegate to the OS. Simultaneous would just be like any other OS that exists already.

Also as well, as long as you have a basic Standard C Library (`memset`, etc. excluding file system stuff) then generally things would be fine there.
